src/nvim/main.c

```diff
@@ -1494,9 +1498,16 @@ static void init_params(mparm_T *paramp, int argc, char **argv)
 /// Initialize global startuptime file if "--startuptime" passed as an argument.
 static void init_startuptime(mparm_T *paramp)
 {
+  bool is_embedded_process = false;
+  for (int i = 1; i < paramp->argc - 1; i++) {
+    if (STRICMP(paramp->argv[i], "--embed") == 0) {
+      is_embedded_process = true;
+      break;
+    }
+  }
   for (int i = 1; i < paramp->argc - 1; i++) {
     if (STRICMP(paramp->argv[i], "--startuptime") == 0) {
-      time_fd = fopen(paramp->argv[i + 1], "a");
+      time_init(paramp->argv[i + 1], embedded_mode ? "Embedded" : "Primary/TUI");
```

Comment:
Is "primary" used anywhere else? Don't think that will help anyone by introducing extra jargon.
```suggestion
      time_init(paramp->argv[i + 1], embedded_mode ? "Embedded" : "TUI");
```

Reply:
yes, I was very unsure on how to call the processes in the report. My first approach was to name "TUI" the client process and "Main" the server, however @zeertzjq correctly realized that the name "TUI" breaks the moment nvim is invoked in headless mode. That's why I started to think in more general terms like, "primary" or "secondary" process, to account for the headless scenario.

Reply:
`TUI/client` addresses that edge-case.

Reply:
I don't think it's "client" as headless instance are usually servers.

Reply:
changed